tests/integration/test_charm.py

```diff
@@ -23,24 +25,45 @@
 async def test_build_and_deploy(
     ops_test: OpsTest,
     charm_base: str,
-    filesystem_client_charm,
-    nfs_server_proxy_charm,
-    cephfs_server_proxy_charm,
+    filesystem_client_charm: Awaitable[str | pathlib.Path],
+    nfs_server_proxy_charm: Awaitable[str | pathlib.Path],
+    cephfs_server_proxy_charm: Awaitable[str | pathlib.Path],
```

Comment:
Are we just using `pathlib.Path` here for the function annotation? If so, why don't we either directly import `Path` &mdash; `from pathlib import Path` &mdash; so that we don't import the entire `pathlib` module, or use [`PathLike`](https://docs.python.org/3/library/os.html#os.PathLike) abstract class from the `os` module.

`PathLike` might make the most sense here if:

1. This annotation is the only place where we're using `Path` in the _test_charm.py_ file, and
2. We'll accept any object that looks and feels like a filesystem path that doesn't need all the methods a `Path` object provides.

Reply:
I'd say to just import `Path`. We already control the type because those come from `conftest.py`, so abstracting around that makes less sense IMO.

Reply:
Works for me. At the end of the day we're just making things easier for ourselves since pyright only checks what is under `src/**` :sweat_smile: